### PR TITLE
chore(build): introduce bundle in Web Components

### PR DESCRIPTION
**Pull request by asudoh**
### Related Ticket(s)

Refs #3980.

### Description

Introduces `ibmdotcom-web-components-dotcom-shell.js` and `ibmdotcom-web-components-dotcom-shell.min.js` Rollup bundles so that our users can quickly set up dotcom shell without setting up any toolstack. Also introduces an example that demonstrates how to use the bundle.

### Changelog

**New**

- `ibmdotcom-web-components-dotcom-shell.js` and `ibmdotcom-web-components-dotcom-shell.min.js` Rollup bundles.
- An example that demonstrates how to use the bundle.

**Changed**

- Changes so our components stop rendering core `carbon-web-components` directly:
  - Replaces `<bx-search>` with `<dds-search>`. The latter is effectively the alias of the former
  - Replaces `<bx-header>` with plain `<header>`
- Replaces `clean-css` with `cssnano` so the minifier is well integrated to other PostCSS toolchain.
- Adds `@babel/plugin-transform-runtime` back so the same Babel helpers won't be repeated in the bundle.
- Integrated `yarn clean` task into `gulp clean` task.
- Changed how `lodash-es` modules are imported for the sake of less reliance to tree-shaking.

**Removed**

### CODE REVIEW

File: packages/web-components/examples/static-files/index.html [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+        margin-top: 0;
+      }
+    </style>
+    <script type="module">
+      // Copy from `dist` and put it to the same directory as this file
+      import './ibmdotcom-web-components-dotcom-shell.min.js';
+    </script>
+  </head>
+  <body>
+    <dds-masthead>
```
Comment (jeffchew) [inline packages/web-components/examples/static-files/index.html:40]: @asudoh does the bundle also include `dds-dotcom-shell-container`? That should probably be the `quick start` version of this, followed by the customized `masthead` for more advanced implementations.

Comment (asudoh) [reply packages/web-components/examples/static-files/index.html:40]: @jeffchew Yes. I started with manual approach due to `CORS_PROXY` thing, van update this once our fix to its reliance us in.

Comment (jeffchew) [reply packages/web-components/examples/static-files/index.html:40]: You can probably already get started on switching over, the access control policy is already live on EI server.

Comment (asudoh) [reply packages/web-components/examples/static-files/index.html]: 🎉 - Updated the examples.

**Review by jeffchew — APPROVED** [review]
Thank you @asudoh , looks good!

**Pull request merged**